### ISSUE TITLE
Performance gap compared with the official TensorFlow implementation

### ISSUE THREAD

**Issue by laoreja** [rosinality/stylegan2-pytorch#91]
Hi,

Thanks for the great repo!

I saw that you mentioned "I have trained the 256px model on FFHQ 550k iterations. I got FID about 4.5. Maybe data preprocessing, resolution, training loop could made this difference, but currently I don't know the exact reason of FID differences."

May I ask what's the performance of the original TensorFlow repo with the same training setting? From their paper and repo, I only found the FID for 1024x1024 resolution. And I haven't used TensorFlow for many years...

Thank you in advance!

**Comment by rosinality** [comment]
You can get 3.81 FID for ffhq at 256px. (from https://arxiv.org/abs/2006.06676)

**Comment by cientgu** [comment]
> You can get 3.81 FID for ffhq at 256px. (from https://arxiv.org/abs/2006.06676)

Yes, but I found in the tensorflow implementation, there are implementation bugs on compute the FID. They compute FID between 50k generated images at 256*256 and 50k ground truth images at 1024*1024, which is significant lower than compute them all on 256*256. I suggest to use the pytorch-fid(https://github.com/mseitzer/pytorch-fid) on both 50k 256*256 images to get the correct FID score.

**Comment by shoutOutYangJie** [comment]
> > You can get 3.81 FID for ffhq at 256px. (from https://arxiv.org/abs/2006.06676)
> 
> Yes, but I found in the tensorflow implementation, there are implementation bugs on compute the FID. They compute FID between 50k generated images at 256x256 and 50k ground truth images at 1024x1024, which is significant lower than compute them all on 256x256. I suggest to use the pytorch-fid(https://github.com/mseitzer/pytorch-fid) on both 50k 256x256 images to get the correct FID score.

Do you know how to evaluate this repo? I can't find the test code.

**Comment by cientgu** [comment]
> > > You can get 3.81 FID for ffhq at 256px. (from https://arxiv.org/abs/2006.06676)
> > 
> > 
> > Yes, but I found in the tensorflow implementation, there are implementation bugs on compute the FID. They compute FID between 50k generated images at 256x256 and 50k ground truth images at 1024x1024, which is significant lower than compute them all on 256x256. I suggest to use the pytorch-fid(https://github.com/mseitzer/pytorch-fid) on both 50k 256x256 images to get the correct FID score.
> 
> Do you know how to evaluate this repo? I can't find the test code.

just use pytorch-fid(https://github.com/mseitzer/pytorch-fid)
